base/Enums.jl

```diff
@@ -84,27 +90,34 @@ macro enum(T,syms...)
             lo = min(lo, i)
             hi = max(hi, i)
         end
+        i += one(i)
     end
-    values = Int32[i[2] for i in vals]
+    values = basetype[i[2] for i in vals]
     if hasexpr && values != unique(values)
         throw(ArgumentError("values for Enum $typename are not unique"))
     end
+    names = Symbol[i[1] for i in vals]
     blk = quote
         # enum definition
-        Base.@__doc__(bitstype 32 $(esc(T)) <: Enum)
+        Base.@__doc__(bitstype $(basetype.size * 8) $(esc(typename)) <: Enum)
+        Base.convert(::Type{Integer}, x::$(esc(typename))) = box($(basetype), x)
+        Base.convert{T<:Integer}(::Type{T}, x::$(esc(typename))) = convert(T, box($(basetype), x))
+        Base.write(io::IO, x::$(esc(typename))) = write(io, $(basetype)(x))
+        Base.read(io::IO, ::Type{$(esc(typename))}) = $(esc(typename))(read(io, $(basetype)))
```

Comment:
The earlier definitions of convert, read, and write in this file should be removed.